oxide-auth/src/primitives/generator.rs

```diff
@@ -132,24 +132,24 @@ impl Assertion {
     /// padding or shortening of the supplied key material may be applied in the form dictated by
     /// the signature type. See the respective standards.
     ///
-    /// If future suites are added where this is not possible, his function may panic when supplied
+    /// If future suites are added where this is not possible, this function may panic when supplied
     /// with an incorrect key length.
+    /// 
+    /// Currently, the implementation lacks the ability to really make use of another hasing mechanism than
+    /// hmac + sha256.
     pub fn new(kind: AssertionKind, key: &[u8]) -> Self {
-        let key = match kind {
-            AssertionKind::HmacSha256 => SigningKey::new(&SHA256, key),
+        match kind {
+            AssertionKind::HmacSha256 => Assertion { key: key.to_vec() },
             AssertionKind::__NonExhaustive => unreachable!(),
-        };
-
-        Assertion {
-            secret: key,
         }
     }
 
     /// Construct an assertion instance whose tokens are only valid for the program execution.
     pub fn ephemeral() -> Self {
-        Assertion {
-            secret: SigningKey::generate(&SHA256, &SystemRandom::new()).unwrap(),
-        }
+        // XXX Why 64?
```

Comment:
There should be a definition on the key size required by hmac-sha256. This will explain the size, I'm currently confused as to why this is not `32`.

Reply:
Oh yeah, cause I'm dumb. I was intending to find how to get to `Mac::KeySize` for a given impl (say `Hmac::<Sha256>`), but I'm not proficient enough in rust to figure it out and I forgot to get back to it.